support/build/Makefile.rules

```diff
@@ -483,19 +483,19 @@ endef
 define buildrule_S_ =
 $(4): $(2) | preprocess
 	$(call build_cmd_fixdep,AS,$(1),$(4),\
-		$(AS)  $$(COMPFLAGS) $$(COMPFLAGS-y) \
-		       $$($(call vprefix_lib,$(1),COMPFLAGS)) $$($(call vprefix_lib,$(1),COMPFLAGS-y)) \
-		       $$(ASINCLUDES) $$(ASINCLUDES-y) \
-		       $$($(call vprefix_lib,$(1),ASINCLUDES)) $$($(call vprefix_lib,$(1),ASINCLUDES-y)) \
-		       $$($(call vprefix_src,$(1),$(2),$(3),INCLUDES)) $$($(call vprefix_src,$(1),$(2),$(3),INCLUDES-y)) \
-		       $$($(call vprefix_glb,$(3),ARCHFLAGS)) $$($(call vprefix_glb,$(3),ARCHFLAGS-y)) \
-		       $$(ASFLAGS) $$(ASFLAGS-y) $$(ASFLAGS_EXTRA) \
-		       $$($(call vprefix_lib,$(1),ASFLAGS)) $$($(call vprefix_lib,$(1),ASFLAGS-y)) \
-		       $$($(call vprefix_src,$(1),$(2),$(3),FLAGS)) $$($(call vprefix_src,$(1),$(2),$(3),FLAGS-y)) \
-		       $(5) \
-		       $$(DBGFLAGS) $$(DBGFLAGS-y) \
+		$(AS)  $(5) \
```

Comment:
Is there an advantage to moving this at the beginning and not leave it where it previously was?

Reply:
So the arguments in these have 2 variants in the original: single-dollar (`$(5)`) and double-dollar (`$$(COMPFLAGS)`). The single-dollar ones will be expanded by make and their contend will also passed to fixdep. The double-dollar will not be expanded and only the name (`$(COMPFLAGS)`) will be passed to `build_cmd_fixdep` and eventually fixdepth.

In the original code, these variables will be resolved before invoking bash (or by bash, I never exactly figured it out. They will not by resolved before writing them into the build_args file.
These  double-dollar variables are exactly the one we want in the build_args file, so we need them to be resolved and put to the new `$longarg` argument of `build_cmd[_fixdepth]`.

Reply:
I remember there was some talk about the need for secondary expansions, Simon should know more.

Reply:
I removed the second commit now and will open a draft PR for that. The first one is enough for now.